maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/VersionsHelper.java

```diff
@@ -80,11 +81,15 @@ private void sortDescending(List<ArtifactVersion> versions){
      * @return inferred version for given artifact
      */
     public String inferVersion(Artifact artifact){
-        List<ArtifactVersion> artifactVersions = getVersions(artifact);
-        if(artifactVersions.size() == 0){
+        if (new Version(artifact.getVersion()).isSnapshot()) {
+            List<ArtifactVersion> artifactVersions = getVersions(artifact);
+            if(artifactVersions.size() == 0){
+                return artifact.getVersion();
+            }
+            return inferVersion(artifact.getVersion(), artifactVersions);
```

Comment:
@rkorytkowski artifact.getVersion is never null(or it shouldn't be :) ). If it was there would be an exception in this [line](https://github.com/openmrs/openmrs-sdk/blob/master/maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/VersionsHelper.java#L58). The only thing this infer method help us is if there was some beta/alpha release and user specified only version ( 2.3 will be changed e.g to 2.3-alpha if 2.3 was not released).  Is there a possibility that there will be some alpha-SNAPSHOT version ? If no then this method is useless now. Let me know and I will apply changes